File: analyzer/src/ast/access_specifier.rs
use crate::{error::CompilerError, lexer::token_type::TokenType, parser::Parser};

use super::AstParse;

#[derive(Debug, Clone)]
pub enum AccessSpecifier {
    Public,
    Private,
}

impl AstParse for AccessSpecifier {
    fn parse(parser: &mut Parser) -> Result<Self, CompilerError> {
        if parser.current().token_type() == &TokenType::Pub {
            parser.next_token()?;
            return Ok(Self::Public);
        }
        return Ok(Self::Private);
    }
}

File: src/Parser/AccessSpecifier.cpp
#include "Ast/AccessSpecifier.hpp"
#include "Parser/Parser.hpp"

namespace Marble
{
    AccessSpecifier ParseAccessSpecifier(Parser &parser, Span &span)
    {
        if (parser.Current().TokenType() != TokenType::Pub)
        {
            return AccessSpecifier::Private;
        }
        span = parser.Current().Span();
        parser.NextToken();
        return AccessSpecifier::Public;
    }
} // namespace Marble
